### PR TITLE
[CARBONDATA-1510] UDF test case added

### PR DESCRIPTION
**Pull request by rahulforallp**
UDF is already working fine . So this PR contains only the test case  for UDF with complex data.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/437/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/312/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1066/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/315/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/440/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1069/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/udf/TestUdf.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
+package org.apache.carbondata.spark.testsuite.udf
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.test.util.QueryTest
+import org.apache.spark.sql.types.{DataType, StringType, StructField, StructType}
+import org.scalatest.BeforeAndAfterAll
+
+class TestUdf extends QueryTest with BeforeAndAfterAll {
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/udf/TestUdf.scala]: Can you rename to mention it is for complex type UDF test?

File: integration/spark-common-test/pom.xml [hunk 2]
```diff
@@ -116,7 +116,6 @@
   </dependencies>
 
   <build>
-    <testSourceDirectory>src/test/scala</testSourceDirectory>
```
Comment (jackylk) [inline integration/spark-common-test/pom.xml]: Is it required to remove this?

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1178/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/547/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/udf/TestUdf.scala (hunk 1 above)
Comment (rahulforallp) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/udf/TestUdf.scala]: @jackylk rename done

File: integration/spark-common-test/pom.xml (hunk 2 above)
Comment (rahulforallp) [reply integration/spark-common-test/pom.xml]: I have removed this because I have added java file . And that java file also need to be compiled.

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1280/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/652/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/724/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1361/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/807/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1440/

**Comment by rahulforallp** [comment]
@jackylk review comment resolved . please review and merge.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1415/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3064/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4308/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5196/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3986/

**Pull request closed without merging**